src/search/merge_and_shrink/labels.h

```diff
@@ -4,45 +4,65 @@
 #include <memory>
 #include <vector>
 
-namespace utils {
-class LogProxy;
-}
-
 namespace merge_and_shrink {
-class Label {
-    /*
-      This class implements labels as used by merge-and-shrink transition systems.
-      Labels are opaque tokens that have an associated cost.
-    */
-    int cost;
+class LabelsConstIterator {
+    const std::vector<int> &label_costs;
+    std::size_t current_index;
+
+    void next_valid_index();
 public:
-    explicit Label(int cost_)
-        : cost(cost_) {
+    LabelsConstIterator(const std::vector<int> &label_costs, bool end);
+    void operator++();
+
+    int operator*() const {
+        return int(current_index);
```

Comment:
I think C-style cases are discouraged, and we should use something like `static_cast`?